algorand/Dockerfile.build

```diff
@@ -1,15 +1,16 @@
 FROM docker.io/python:3.10@sha256:eeed7cac682f9274d183f8a7533ee1360a26acb3616aa712b2be7896f80d8c5f as algorand-contracts-build
 
-RUN python3 -m pip install virtualenv
-COPY Pipfile.lock Pipfile.lock
-COPY Pipfile Pipfile
+RUN mkdir algosdk
+RUN --mount=type=cache,target=/algosdk wget https://github.com/algorand/go-algorand/releases/download/v3.19.0-stable/node_stable_linux-amd64_3.19.0.tar.gz --output-document=algosdk/node-stable-linux.tar.gz
+RUN --mount=type=cache,target=/algosdk tar --extract --file algosdk/node-stable-linux.tar.gz bin/goal && rm algosdk/node-stable-linux.tar.gz
+
+RUN mkdir teal
+COPY Pipfile Pipfile.lock *.py ./
```

Comment:
why move the `COPY *.py` before the `pipenv install` command? this way, each time any of the python files are modified, `pipenv install` will be run again. The old version only reran `pipenv install` when the `Pipfile` changed

Reply:
Hm, you're right. Once again, I was running into something that was triggering a rebuild always but it was probably me modifying the dockerfile. I can no longer reproduce it after reintroducing the copy command at least.